### ISSUE TITLE
[Accessibility Docs] Muted heading

### ISSUE THREAD

**Issue by sowasred2012** [canonical/vanilla-framework#4056]
Using [this template](https://docs.google.com/document/d/1IwFMfJrTfJPXDcVJfLWXYkw-pjgWhBWo4GX8X098Q64/edit#), create accessibility docs for [Muted heading](https://vanillframework.io/docs/patterns/muted-heading)

**Comment by bartaz** [comment]
Triaging: High effort (docs don't exist), impacts accessibility (documentation only).
Should be addressed in next architecture docs. Not doing as part of v4.
